Modern/ops/src/test/java/org/bonej/ops/ellipsoid/FindEllipsoidFromBoundaryPointsTest.java

```diff
@@ -0,0 +1,224 @@
+package org.bonej.ops.ellipsoid;
+
+import net.imagej.ImageJ;
+import net.imagej.ops.special.function.BinaryFunctionOp;
+import net.imagej.ops.special.function.Functions;
+import net.imglib2.util.ValuePair;
+import org.joml.Matrix3d;
+import org.joml.Matrix4d;
+import org.joml.Vector3d;
+import org.joml.Vector4d;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ *Tests for {@link FindEllipsoidFromBoundaryPoints}
+ *lots of expected values calculated with pen and paper or sympy.
+ *TODO: put calculations in javadoc and user documentation
```

Comment:
I agree

Reply:
Also asserts could have messages explaining what went wrong in more detail

Reply:
You should also have a test to check that finding the ellipsoid fails when it should (`assertFalse(ellipsoid.isPresent())`)

Reply:
Could you use any of the examples in the article as test cases?